### PR TITLE
Add CI Test Matrix

### PR DESCRIPTION
**Pull request by Sunnatillo**
This PR adds CI Test Matrix, describing which branches and image tags are used in CI tests

### CODE REVIEW

File: docs/user-guide/src/version_support.md [hunk 1]
```diff
@@ -53,4 +53,26 @@ Supported container images for BMO and Ironic can be found in quay. Examples are
...
-- quay.io/metal3-io/ip-address-manager:v1.2.0
\ No newline at end of file
+- quay.io/metal3-io/ip-address-manager:v1.2.0
+
+## CI Test Matrix
+
+The table describes which branches/image-tags are tested in each periodic CI tests:
+
+<!-- markdownlint-disable -->
+| Test                                                   | CAPM3 branch | IPAM branch  | BMO branch/tag  | Keepalived tag | MariaDB | VBMC   | Sushi-tools | Ironic |
```
Comment (furkatgofurov7) [inline docs/user-guide/src/version_support.md:63]:
```suggestion
| Test                                                   | CAPM3 branch | IPAM branch  | BMO branch/tag  | Keepalived tag | MariaDB | VBMC   | sushy-tools | Ironic |
```

File: docs/user-guide/src/version_support.md [hunk 2]
```diff
@@ -53,4 +53,26 @@ Supported container images for BMO and Ironic can be found in quay. Examples are
...
+| ------------------------------------------------------ | ------------ | -----------  | --------------- | -------------- | ------- | ------ | ----------- | ------ |
+| INTEGRATION TESTS                                      |              |              |                 |                |         |        |             |        |
+| daily_main_integration_test_ubuntu/centos              | main         | main         | main            | latest         | latest  | latest | latest      | latest |
+| daily_main_e2e_integration_test_ubuntu/centos          | main         | main         | main            | latest         | latest  | latest | latest      | latest |
+| daily_release-1-3_integration_test_ubuntu/centos       | release-1.3  | release-1.3  | v0.2.0          | v0.2.0         | latest  | latest | latest      | latest |
+| daily_release-1-1_integration_test_ubuntu/centos       | release-1.2  | release-1.2  | v0.1.2          | v0.1.2         | latest  | latest | latest      | latest |
```
Comment (furkatgofurov7) [inline docs/user-guide/src/version_support.md:69]:
```suggestion
| daily_release-1-2_integration_test_ubuntu/centos       | release-1.2  | release-1.2  | v0.1.2          | v0.1.2         | latest  | latest | latest      | latest |
```

File: docs/user-guide/src/version_support.md [hunk 3]
```diff
@@ -53,4 +53,26 @@ Supported container images for BMO and Ironic can be found in quay. Examples are
...
+| daily_release-1-2_integration_test_ubuntu/centos       | release-1.1  | release-1.1  | v0.1.1          | v0.1.1         | latest  | latest | latest      | latest |
```
Comment (furkatgofurov7) [inline docs/user-guide/src/version_support.md:70]:
```suggestion
| daily_release-1-1_integration_test_ubuntu/centos       | release-1.1  | release-1.1  | v0.1.1          | v0.1.1         | latest  | latest | latest      | latest |
```

**Review by furkatgofurov7 — CHANGES_REQUESTED** [review]
The doc itself is pretty outdated as well, but maybe that is out of the scope for this change

File: docs/user-guide/src/version_support.md (hunk 2 above)
Comment (Sunnatillo) [reply docs/user-guide/src/version_support.md]: done

File: docs/user-guide/src/version_support.md (hunk 1 above)
Comment (Sunnatillo) [reply docs/user-guide/src/version_support.md]: done

[the remark this answers is not in the document]
Comment (Sunnatillo) [reply docs/user-guide/src/version_support.md]: I tried several ways, as you can check on different commits. Solution works for Vscode preview, but changes did not work on github. I am out of ideas.

**Review by tuminoid — COMMENTED** [review]
LGTM. The doc in general needs an update in another PR, but the testing matrix seems fine. It doesn't render well in Github docs, but given the width and amount of information, I don't think that can be helped.

**Comment by tuminoid** [comment]
/assign @furkatgofurov7 @kashifest 

Let's get this merged.

File: docs/user-guide/src/version_support.md [hunk 4]
```diff
@@ -53,4 +53,27 @@ Supported container images for BMO and Ironic can be found in quay. Examples are
...
+- quay.io/metal3-io/ip-address-manager:v1.2.0
+
+## CI Test Matrix
+
+The table describes which branches/image-tags are tested in each periodic CI tests:
+
+<!-- markdownlint-disable -->
+
+| TESTS                                                | CAPM3 branch | IPAM branch  | BMO branch/tag  | Keepalived tag | MariaDB | VBMC   | sushi-tools | Ironic |
+| ------------------------------------                 | ------------ | -----------  | --------------- | -------------- | ------- | ------ | ----------- | ------ |
```
Comment (dtantsur) [inline docs/user-guide/src/version_support.md:65]: While we may want to eventually version Ironic and MariaDB, I'd not envision versioning vbmc and sushy-tools (you have a typo in the name btw).

Comment (Sunnatillo) [reply docs/user-guide/src/version_support.md]: Thank for reviewing. I addressed your comments,  Removed sushy-tools and VBMC from the table, added line mentioning we use latest versions of them.

**Review by tuminoid — COMMENTED** [review]
Renders bit better now. I think all comments are addressed.

/lgtm

**Comment by kashifest** [comment]
Lets update the matrix once more with new release branch and tags ?

**Comment by tuminoid** [comment]
> Lets update the matrix once more with new release branch and tags ?

Can we do it in another PR, so I can use that PR as an example PR in release documentation?

**Pull request merged**